### DISCUSSION TITLE
1.8" 160x128 st7735s boots to white backlight

### DISCUSSION THREAD

**Discussion by sej7278** [Bodmer/TFT_eSPI#998] [Q&A - General] [answered]
Does anyone know why this module (looks like an [adafruit clone](https://www.adafruit.com/product/618)) boots to a white screen with a backlight on?

![img](https://i.imgur.com/KL4tWJ9.jpg)

The 7735 isn't even supposed to have backlight control but `digitalWrite(TFT_BL, LOW);` turns it off and `digitalWrite(TFT_BL, HIGH);` turns it on just fine - its even got a BLK pin!

If i turn it off, then i get no display at all, so its not like a backlight to me, more like a screen on/off switch, so doubt i could just wire the pin to GND or disconnect it.

That said, the 7789 on the TTGO Display doesn't seem to do anything when you toggle the backlight, so maybe my info is backwards?

Is there something I have to do before/after `tft.init()` perhaps? i've tried swapping `#define TFT_BACKLIGHT_ON HIGH` around.

on a completely different note (!!!) the digital font 7 seems to have weird kerning - like "17:02" will have a large space to the left, like it has no kerning at all on the "1", it it monospaced? is there anything i can do about that - padding or something to set, or do I need to make a vfw version of the font? i noticed is tft.fontWidth() never changes - "88:88" is the same width as "11:11".

**Comment by Bodmer** [comment]
The ST7735 is the display pixel driver chip and has no backlight control built in.  The white LEDs are an independant electrical circuit, some displays sometimes come with a transistor to turn on/off the LEDs, others need to have VCC (3.3V or 5V) connected to the pin as a digital I/O pin would be overloaded. Check your display has a transistor, typically this is visible on the back of the board. The backlight LEDs have no function other that to shine through the LCD panel so you can see the pixels.

TTGO produce quite a few display boards, it is not clear which one you are referring to. Some have  a backlight control pin, others do not. Check the vendors web page for the technical details and features.

I am not sure what you are asking, guessing - if you what to control the backlight from you sketch then you can use digitalWrite.

Font 7 emulates a 7 segment display and the character spacing associated with that type of display.  Incidentally monspaced fonts for digits are an advantage, otherwise the digit positions change with the value displayed and this "jiggling" is visually bad.

**Comment by sej7278** [comment]
i guess what i'm asking is:

1. are the white led's needed and will i see any display without them? thinking of disconnecting BLK.
2. why do they default to on - is that the library or just a default state before the sketch has loaded?

the BLK pin goes to a transistor via a resistor yes.

its just odd that the ttgo-t-display and another display i bought (240x240 1.3" st7789) don't have this "white flash" on boot and neither seems to do anything when you digitalWrite() the backlight pin, i wish the 7735 display worked the same!

good point about the monospaced LCD font, i may give it a different background colour to the main background so it looks like its within a fixed square and not just too right-aligned.

**Comment by Bodmer** [comment, marked as answer]
1. Yes, that is how these displays work. Under bright light you might see a very faint image under bright lights but it will not be usable.
2. A design decision for the default by the original designer. You could change this by putting a suitable value pull-down resistor on the pin. Some displays have a backlight off by default (pull down on base of NPN transistor).

Try [this example ](https://github.com/Bodmer/TFT_eSPI/tree/master/examples/160%20x%20128/TFT_Clock_Digital)and see line 97

**Comment by sej7278** [comment]
i think i'm being dumb here, as i can't understand how one 7735 board has a bright white screen on boot when two 7789 boards have black screens, but they obviously all have the backlights on or you wouldn't see anything (and i'm not turning on the backlight in the 7789 board code, its just on i guess).

are the 7789's perhaps pwm dimming the backlight by default hence why it doesn't seem to make any difference when you toggle it - unlike the 7735 which definitely goes fully on or fully off when you toggle it? or just dimmer LED's?

can't really see much about the TTGO T-Display's actual display in their info [here](http://www.lilygo.cn/prod_view.aspx?TypeId=50044&Id=1126&FId=t3:50044:3) the don't even mention the 7789 in the schematic, let alone the screen!

the 1.8" and 1.3" boards i got are from [aliexpress](https://www.aliexpress.com/item/4001144194129.html)

**Comment by Bodmer** [comment]
The LED backlights have no electrical connection to the ST7735 or the ST7789 chips. They are contolled by the user or are hardwired on.

The information can be difficult to find on the TTGO T disiplay. The backlight is on GPIO 4:

![image](https://user-images.githubusercontent.com/15824805/107814452-cf30af00-6d69-11eb-89f4-f1553a7c1e80.png)

**Comment by Bodmer** [comment]
The TTGO schematic does have the display shown (P1 interface) and the backlight transistor (Q0) shown. The transistor is biased off by default:
![image](https://user-images.githubusercontent.com/15824805/107816243-63037a80-6d6c-11eb-9ec2-b52b23ea0fd6.png)
